js1-week2-homework/assets/operators.js

```diff
@@ -0,0 +1,51 @@
+// 1:
+0 == '0'; // true
+0 == []; // true
+'0' == []; // false
+/*-- 
+Line 1: 0 is falsy value, '0' is not. 
+        JS will coerce the types when using double equals method. 
+        That means it will coerce both value to 0 = 0 wich is true.
+Line 2: The empty array is truthy, 0 is falsy. 
+        JS will coerce both to numbers. The empty array lenght is 0 
+        so the result will be ture as 0 == 0
+line 3: Both valus are truthy. JS can't coerce them to become the same type
+        therfor the resul is false */
```

Comment:
Ahaan - actually when we compare `'0' == []` or `0 == []` - JavaScript will try to convert the empty array to a string using `toString` i.e [].toString() which is equal to an empty string. Thus, `0 == []` is essentially comparing `0 == ""` which is true since both sides are falsy. However, `'0' == []` results in `'0' == ''` which is false since right side is falsy but left side is truthy. Does it make sense? 🙂

Reply:
Have a look here as well - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Comparison_Operators

Reply:
Thanks Yash! That make sense to me.